modules/kok-bot/modules/karma/listener.js

```diff
@@ -10,45 +11,63 @@ client.on("message", msg => {
 
         var reply = new Array();
 
-        mentioned.array().forEach((mention, index, mentions) => {
-            if (mention.id === msg.author.id) {
-                reply.push(`Trying to give yourself karma ${mention.username}. Shame!`);
-            } else {
-                mentionUsers.push(mention.username);
-                let guildID = msg.guild.id;
-                guildUsersModel.findOneOrCreate({ guildID }, { guildID })
-                    .then(guildUsers => {
-                        if (guildUsers) {
-                            let duplicate = false;
-                            guildUsers.users.forEach((user, index, users) => {
-                                if (user.id === mention.id) {
-                                    if (!users[index].karma) {
-                                        users[index].karma = 0;
+        guildUsersModel.findOne({ guildID: msg.guild.id, "users.id": msg.author.id }, { users: { $elemMatch: { id: msg.author.id } } })
+            .then(guildUsers => {
+                console.log(guildUsers);
```

Comment:
Need to do a check for guildUsers as it might return undefined here. Plus remove the console.log from production please.

I suggest
`if (guildUsers && guildUsers.users.length)`